### PR TITLE
cpp: Send clangd compilationDatabasePath in 'initialize' request

### PR DESCRIPTION
**Pull request by MarkZ3**
That way, as soon as ths initialize is received by the server, it can start
indexing with a valid compilation database and not have to wait for a
'didChangeConfiguration' that might or might not happen.

I was tempted to keep the initial "didChangeConfiguration" so that it doesn't
break existing Clangd but there was no released versions yet with this support
anyway. So, provided that we can change Clangd before next version (branching)
then we don't have to keep this compatibility bit.

Depends on https://reviews.llvm.org/D49833

Signed-off-by: Marc-Andre Laperle <marc-andre.laperle@ericsson.com>

### CODE REVIEW

File: packages/cpp/src/browser/cpp-language-client-contribution.ts [hunk 1]
```diff
@@ -27,6 +27,13 @@ import { CPP_LANGUAGE_ID, CPP_LANGUAGE_NAME, HEADER_AND_SOURCE_FILE_EXTENSIONS }
 import { CppBuildConfigurationManager, CppBuildConfiguration } from "./cpp-build-configurations";
 import { CppBuildConfigurationsStatusBarElement } from "./cpp-build-configurations-statusbar-element";
 
+// Clangd extension to set clangd-specific "initializationOptions" in the
```
Comment (simark) [inline packages/cpp/src/browser/cpp-language-client-contribution.ts]: I think we use jsdoc comments (`/** ... */`) to document types.

File: packages/cpp/src/browser/cpp-language-client-contribution.ts [hunk 2]
```diff
@@ -52,21 +59,22 @@ export class CppLanguageClientContribution extends BaseLanguageClientContributio
...
-        this.onActiveBuildConfigChanged(this.cppBuildConfigurations.getActiveConfig());
         this.cppBuildConfigurations.onActiveConfigChange(config => this.onActiveBuildConfigChanged(config));
 
         // Display the C/C++ build configurations status bar element to select active build config
         this.cppBuildConfigurationsStatusBarElement.show();
     }
 
+    private createClangdConfigurationParams(config: CppBuildConfiguration | undefined): ClangdConfigurationParamsChange {
+        const clangdParams: ClangdConfigurationParamsChange = {
+            compilationDatabasePath: config ? config.directory : undefined
```
Comment (simark) [inline packages/cpp/src/browser/cpp-language-client-contribution.ts]: See PR #2437, I think we'll want to use `""` instead of `undefined`.

Comment (MarkZ3) [reply packages/cpp/src/browser/cpp-language-client-contribution.ts]: We'll have to make sure using "" doesn't disable the fallback "check in the parent" algorithm.

File: packages/cpp/src/browser/cpp-language-client-contribution.ts (hunk 1 above)
Comment (MarkZ3) [reply packages/cpp/src/browser/cpp-language-client-contribution.ts]: done

**Comment by simark** [comment]
Hmm there is a conflict, I'll open a new PR since I can't update this one myself.

**Pull request closed without merging**